### PR TITLE
Time picker, timezone support, time mask tokens

### PR DESCRIPTION
**Pull request by nathanreyes**
This is a draft PR for review and testing.

## Todos
* `input` testing
* Localize numeric hour and minute options
* Convert `TimePicker.vue` to a renderless component for user customization via `time-picker` scoped slot
* Update docs and changelog

### CODE REVIEW

File: src/utils/fecha.js [hunk 1]
```diff
@@ -234,8 +237,11 @@ parseFlags.mm = parseFlags.m;
 parseFlags.hh = parseFlags.H = parseFlags.HH = parseFlags.h;
 parseFlags.ss = parseFlags.s;
 parseFlags.A = parseFlags.a;
+parseFlags.XXX = parseFlags.XX = parseFlags.X;
 
 export const format = (dateObj, mask, locale) => {
+  mask = (isArray(mask) && mask) || (isString(mask) && mask) || 'YYYY-MM-DD';
```
Comment (DavidRouyer) [inline src/utils/fecha.js:243]: https://github.com/nathanreyes/v-calendar/blob/a806fb8334cdb1c6c61ef0ddacb6f2db9f298046/src/utils/fecha.js#L254

`mask.replace` fails when mask is an array. `TypeError: mask.replace is not a function`

See the "Popover Mode" example on the homepage

**Comment by DavidRouyer** [comment]
Do you have more recent changes on https://github.com/nathanreyes/v-calendar/tree/feature/timepicker?
What should I review?

I haven't seen a prop to enable/disable the timepicker on `v-date-picker` ?

**Comment by nathanreyes** [comment]
@DavidRouyer The `mode` prop is going to be used to switch between `date` (default), `dateTime` and `time` modes. There will be a new `is-range` prop to bind to date ranges.

**Comment by nathanreyes** [comment]
For further details, you can run `yarn dev` to explore the docs for all of the api updates. Still need to tidy up a few things.

**Comment by nathanreyes** [comment]
All changes merged into `next` branch.

**Pull request closed without merging**